src/js/controller/instream-html5.js

```diff
@@ -161,7 +159,10 @@ const InstreamHtml5 = function(_controller, _model) {
             return;
         }
 
-        var isVpaidProvider = provider.type === 'vpaid';
+        // Let the player media model know we're using it's video tag
+        _model.mediaModel.srcReset();
```

Comment:
Check provider is also called by apply listeners. It should not reset the media model. Only instream load(), and external clearing of the video.src should reset the mediaModel.

You should listen for video “unload” event to determine that a new src was loaded.

You could also move the source comparison code in html5 into a module and use it to compare the item source to to the video on destroy to determine if we need to do a reset outside of instream load.